tests/test_nn.py

```diff
@@ -770,7 +770,9 @@ def test_multihead_attention(getkey):
     q = jrandom.uniform(getkey(), (19, 3))
     k = jrandom.uniform(getkey(), (23, 5))
     v = jrandom.uniform(getkey(), (23, 7))
-    assert attn(q, k, v).shape == (19, 11)
+    out = attn(q, k, v)
+    assert isinstance(out, jnp.ndarray)
+    assert out.shape == (19, 11)
```

Comment:
As above, it'd be great to have some tests for all this new functionality!

Reply:
I've just written a little test using the LLaMA implementation and our implementation and got the same results (for the KV cache) -- just need to finalise it